### PR TITLE
Enable AvroSchemaManager to read schema from Kafka schema registry

### PR DESCRIPTION
**Pull request by abti**
Enable AvroSchemaManager to read schema from Kafka schema registry if the Uri scheme is http(s) or fall back to HDFS

### CODE REVIEW

**Comment by abti** [comment]
@pcadabam Please review

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java [hunk 1]
```diff
@@ -12,11 +12,14 @@
...
 import java.io.IOException;
+import java.net.URI;
 import java.nio.charset.StandardCharsets;
 import java.util.Map;
-
 import lombok.extern.slf4j.Slf4j;
+
 import org.apache.avro.Schema;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
```
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: Can you update the class javadoc for AvroSchemaManager with this new case?

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java [hunk 2]
```diff
@@ -110,7 +113,22 @@ public void cleanupTempSchemas() throws IOException {
   }
 
   public static Schema getSchemaFromUrl(Path schemaUrl, FileSystem fs) throws IOException {
```
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: I think the argument should not be a ```Path```. It should be changed to a ```URI```
Add javadoc stating the schemaUrl can be with or without scheme and it defaults to hdfs.

Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: Correction: Changing this to ```URI``` will need wider refactor as ```SerializableHivePartition``` and ```SerializableHiveTable``` use ```Path``` as schemaUrl. 
I suggest moving this logic to read from http/https into ```getOrGenerateSchemaFile```. So we would be creating a temp schema file on HDFS by reading the schema from registry. This way we are consistent with other cases and we guarantee schema used in source is same as in extractor.

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java (hunk 1 above)
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: As discussed with change in the logic location, its not needed anymore

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java (hunk 2 above)
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: As discussed moved it to other location

**Comment by pcadabam** [comment]
LGTM

**Pull request merged**